servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/ClientSecurityConfigurator.java

```diff
@@ -51,64 +62,32 @@
      *
      * @param hostNameVerificationAlgorithm The algorithm to use when verifying the host name.
      * See <a href="https://docs.oracle.com/javase/8/docs/technotes/guides/security/StandardNames.html#jssenames">
-     * Supported algorithm names</a>.
+     * Endpoint Identification Algorithm Name</a>.
      * @return {@code this}.
      * @see SSLParameters#setEndpointIdentificationAlgorithm(String)
      */
     ClientSecurityConfigurator hostnameVerificationAlgorithm(String hostNameVerificationAlgorithm);
 
     /**
-     * Determines what algorithm to use for hostname verification.
-     *
-     * @param hostNameVerificationAlgorithm The algorithm to use when verifying the host name.
-     * See <a href="https://docs.oracle.com/javase/8/docs/technotes/guides/security/StandardNames.html#jssenames">
-     * Supported algorithm names</a>.
-     * @param hostNameVerificationHost the host name used to verify the
-     * <a href="https://tools.ietf.org/search/rfc2818#section-3.1">server identity</a>.
-     * @return {@code this}.
-     * @see SSLParameters#setEndpointIdentificationAlgorithm(String)
-     */
-    ClientSecurityConfigurator hostnameVerification(String hostNameVerificationAlgorithm,
-                                                    String hostNameVerificationHost);
-
-    /**
-     * Determines what algorithm to use for hostname verification.
+     * Disable verification of the <a href="https://tools.ietf.org/search/rfc2818#section-3.1">server identity</a>.
```

Comment:
While you are here, can you please clarify in the javadoc the side-effect of this method (i.e. `hostnameVerificationAlgorithm/peerHost/peerPort` will be ignored and it makes your config vulnerable). 

Btw, wdyt about removing this method? For example, grpc-java does not allow users to disable hostname verification. Our generated certificates validate `localhost` hostname, which we can use in tests and examples instead of disabling verification. Users can use our certs for tests or generate their certs with the required hostname.